metals/src/main/scala/scala/meta/internal/decorations/SyntheticsDecorationProvider.scala

```diff
@@ -298,26 +318,147 @@ final class SyntheticsDecorationProvider(
+          explorePatterns(tuplePats)
+        case m.Pat.Bind(lhs, rhs) =>
+          List(lhs.pos.toSemanticdb) ++ explorePatterns(List(rhs))
+        case _ => Nil
+      }
+    }
+
+    def visit(tree: m.Tree): List[s.Range] = {
+      tree match {
+        case vl: m.Defn.Val =>
+          val values =
+            if (vl.decltpe.isEmpty) explorePatterns(vl.pats) else Nil
+          values ++ visit(vl.rhs)
+        case vr: m.Defn.Var =>
+          val values =
+            if (vr.decltpe.isEmpty) explorePatterns(vr.pats) else Nil
+          values ++ vr.rhs.toList.flatMap(visit)
+        case df: m.Defn.Def =>
+          val namePos = df.name.pos.toSemanticdb
+
+          def lastParamPos = for {
+            group <- df.paramss.lastOption
+            param <- group.lastOption
+            token <- param.findFirstTrailing(_.is[T.RightParen])
+          } yield token.pos.toSemanticdb
+
+          def lastTypeParamPos = for {
+            typ <- df.tparams.lastOption
+            token <- typ.findFirstTrailing(_.is[T.RightBracket])
+          } yield token.pos.toSemanticdb
+
+          def lastParen = if (df.paramss.nonEmpty)
+            df.name.findFirstTrailing(_.is[RightParen]).map(_.pos.toSemanticdb)
+          else None
+
+          val values =
+            if (df.decltpe.isEmpty) {
+              val destination =
+                lastParamPos
+                  .orElse(lastTypeParamPos)
+                  .orElse(lastParen)
+                  .getOrElse(namePos)
+              methodPositions += namePos -> destination
+              List(namePos)
+            } else {
+              Nil
+            }
+          values ++ visit(df.body)
+        case other =>
+          other.children.flatMap(visit)
+      }
+    }
+
+    val declarationsWithoutTypes = for {
+      tree <- trees.get(path).toIterable
+      pos <- visit(tree)
+    } yield pos
+
+    val allTypes = declarationsWithoutTypes.toSet
+    val typeDecorations = for {
+      tree <- trees.get(path).toIterable
+      textDocumentInput = Input.VirtualFile(textDocument.uri, textDocument.text)
+      treeInput = Input.VirtualFile(textDocument.uri, tree.pos.input.text)
+      semanticDbToTreeEdit = TokenEditDistance(textDocumentInput, treeInput)
+      treeToBufferEdit = buffer.tokenEditDistance(path, tree.pos.input.text)
+      occ <- textDocument.occurrences
+      range <- occ.range.toIterable
+      treeRange <- semanticDbToTreeEdit.toRevisedStrict(range).toIterable
+      if occ.role.isDefinition && allTypes(treeRange)
+      signature <- textDocument.symbols.find(_.symbol == occ.symbol).toIterable
+      decorationPosition = methodPositions.getOrElse(treeRange, treeRange)
+      realPosition <- treeToBufferEdit.toRevisedStrict(decorationPosition)
+    } yield {
+      val lspRange = realPosition.toLSP
+      signature.signature match {
+        case m: MethodSignature =>
+          val decoration = " : " + decorationPrinter.printType(m.returnType)
```

Comment:
super tiny, but I think it would look better without the leading space (close to the actual code). WDYT?

Reply:
Good idea! will do!

Reply:
changed!